test/filesystemevent/main.d

```diff
@@ -174,8 +172,198 @@ class FileModificationTestTask: Task
                     break;
 
                 default:
-                    test(false, format("Unexpected file system event notification. {}", event));
+                    test(false, "Unexpected file system event notification.");
+            }
+        }
+    }
+}
+
+class FileCreationTestTask: Task
+{
+    /***************************************************************************
+
+        File operations to be checked
+
+    ***************************************************************************/
+
+    private bool created;
+    private bool modified = false;
+    private bool deleted  = false;
+    private bool closed   = false;
+
+    /***************************************************************************
+
+        Name of the created file.
+
+    ***************************************************************************/
+
+    private cstring created_name;
+
+    /***************************************************************************
+
+        Path to the monitored file.
+
+    ***************************************************************************/
+
+    private FilePath temp_path;
+
+    /***************************************************************************
+
+        Variable to control/test the order of the file operations
+
+    ***************************************************************************/
+
+    private int operation_order = 0;
+
+    /***************************************************************************
+
+        Path to the monitored directory.
+
+    ***************************************************************************/
+
+    private cstring watched_path;
+
+    /***************************************************************************
+
+        Tested FileSystemEvent instance.
+
+    ***************************************************************************/
+
+    private FileSystemEvent inotifier;
+
+    /***************************************************************************
+
+        Test entry point. Prepares environment and tests the FileSystemEvent.
+
+    ***************************************************************************/
+
+    override public void run ( )
+    {
+        auto sandbox = DirectorySandbox.create();
+        scope (exit)
+            sandbox.exitSandbox();
+
+        this.watched_path = sandbox.path;
+
+        this.inotifier  = new FileSystemEvent(&this.fileSystemHandler);
```

Comment:
Yeah, I would factor this out into two separate methods. I wanted to use the same callback, so I could test `SmartUnion`, but there's no excuse for not extracting this bit.